lib/Runtime/Library/JsBuiltIn/JsBuiltIn.js

```diff
@@ -415,4 +418,31 @@
         return A;
     });
 
+    platform.registerFunction(FunctionsEnum.ObjectFromEntries, function (iterable) {
+        // #sec-object.fromentries
+        "use strict";
+        if (iterable === null || iterable === undefined) {
+            __chakraLibrary.raiseNeedObject("Object.prototype.fromEntries");
+        }
+
+        const o = {};
+        const propDescriptor = {
+            enumerable : true,
+            configurable : true,
+            writable : true,
+            value : undefined
+        };
+
+        let key;
+        for (const entry of iterable) {
```

Comment:
This can be tainted by overwriting `Array.prototype[Symbol.iterator]`, I believe. Thats why I never used it in Intl, anyways. We should probably check, or use a regular for loop to be safe?

Reply:
Per the proposed spec this method is specifically meant to use the iterator method of the provided iterable - i.e. if using an array and Array.prototype[Symbol.iterator] is overwritten it's meant to behave differently.

Reply:
Oh, weird. All of the Intl things that iterate through arrays call out `for (var x in obj) { if (HasOwnProperty(obj, x) { … } }` or something similar. May want to add a test for tainting the iterator to make sure it _is_ tainted, then, rather than that it isn't.

Reply:
It's because the method is designed to be usable with other iterables, not just arrays e.g. generator functions etc.

Reply:
Yeah, see [DETAILS.md](https://github.com/tc39/proposal-object-from-entries/blob/master/DETAILS.md#iterable-or-array-like) in the proposal.

This is to match the very similar behavior of the [Map constructor](https://tc39.github.io/ecma262/#sec-map-iterable) (which is in fact defined using the same abstract operation).

> May want to add a test for tainting the iterator to make sure it is tainted, then, rather than that it isn't.

There's [one in test262](https://github.com/tc39/test262/blob/5bb0f4c82095ffb9401d3ec83e00c964c923e5ff/test/built-ins/Object/fromEntries/uses-keys-not-iterator.js)! Not sure if you're using that or would consider it sufficient, just thought I'd mention.

Reply:
@bakkot I ran this against test262 before submitting the PR but the Chakracore CI doesn't run test262 so I've added alternate tests for this - check the diff of this PR for tests.